### PR TITLE
DOC: Show and correct default alignment parameters in text.py

### LINKED ISSUE #27345: [Doc]: text alignment defaults
### Documentation Link

https://matplotlib.org/devdocs/api/_as_gen/matplotlib.pyplot.text.html

### Problem

* It would be fantastic if the default text ``horizontalalignment`` (``left``) and ``verticalalignment`` (``baseline``) were documented in ``matplotlib.pyplot.text.html`` please.
* For reference, the defaults are not set in ``rcParams``, but in the ``Text`` class at ``lib/matplotlib/text.py`` here:
```
def __init__(self,
             ...
             verticalalignment='baseline',
             horizontalalignment='left',
             ...
             ):
```
There are a few other similar excerpts in ``text.py`` for resetting defaults, etc.

~~Sorry I'm not skilled enough to do a PR myself.~~ [Got it, I believe: #27346] Thank you for your kind consideration of this request.

### Suggested improvement

* alongside "``horizontalalignment`` or ``ha``" 
    - change: {'left', 'center', 'right'}
    - to: {'left', 'center', 'right'}, default: left
* alongside "``verticalalignment`` or ``va``" 
    - change: {'bottom', 'baseline', 'center', 'center_baseline', 'top'}
    - to: {'bottom', 'baseline', 'center', 'center_baseline', 'top'}, default: baseline

### PR DESCRIPTION
**Pull request by jsalsman**
For issue **[Doc]: text alignment defaults**: closes #27345

## PR summary
Show the default text ``horizontalalignment``/``ha`` (``left``) and ``verticalalignment``/``va`` (``baseline``) in the parameter docstrings (and [the associated gallery page](https://matplotlib.org/devdocs/gallery/text_labels_and_annotations/text_alignment.html)), and correct an error falsely claiming  that ``bottom`` was the vertical default.

## PR checklist
- [X] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [X] *Plotting related* features are demonstrated in https://matplotlib.org/devdocs/gallery/text_labels_and_annotations/text_alignment.html
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [X] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -1002,7 +1002,7 @@ def set_horizontalalignment(self, align):
 
         Parameters
         ----------
-        align : {'left', 'center', 'right'}
+        align : {'left', 'center', 'right'}, default: left
```
Comment (StefRe) [inline lib/matplotlib/text.py:1005]: There is now default value here as the `align` parameter is not optional (i.e. you can't call `set_horizontalalignment()` and expect that it sets the alignement to `left`)

File: lib/matplotlib/text.py [hunk 2]
```diff
@@ -1251,7 +1251,8 @@ def set_verticalalignment(self, align):
 
         Parameters
         ----------
-        align : {'bottom', 'baseline', 'center', 'center_baseline', 'top'}
+        align : {'bottom', 'baseline', 'center', 'center_baseline', 'top'}, \
+default: baseline
         """
```
Comment (StefRe) [inline lib/matplotlib/text.py:1256]: same as above: there is now default value here as the `align` parameter is not optional.

**Comment by ksunden** [comment]
This is a little complicated because these _do_ have a default value in `__init__`, but not in their `set_<attr>` method... but the way the docstring is constructed takes the descriptions from the setters.

I _think_ we may be able to do some appending of default values where that table gets constructed (though I have not looked at all the places that code is used/all the docstrings it interacts with, which could result in confusing double `Default: ...` sections if there already is one). However, it is several layers deep in how it is constructed, so may not be trivial to do so (not the least because you somehow need to thread through where to take the defaults from, and that isn't there now:

- [`_docstring._ArtistPropertiesSubstitution`](https://github.com/matplotlib/matplotlib/blob/43929f31ced180b9ea6c89d25f7ade70d4749b9b/lib/matplotlib/_docstring.py#L64)
- [`_docstring._ArtistKwdocLoader](https://github.com/matplotlib/matplotlib/blob/43929f31ced180b9ea6c89d25f7ade70d4749b9b/lib/matplotlib/_docstring.py#L50)
- [`artist.kwdoc`](https://github.com/matplotlib/matplotlib/blob/43929f31ced180b9ea6c89d25f7ade70d4749b9b/lib/matplotlib/artist.py#L1837)
- [`artist.ArtistInspector`](https://github.com/matplotlib/matplotlib/blob/43929f31ced180b9ea6c89d25f7ade70d4749b9b/lib/matplotlib/artist.py#L1416)
- [`artist.ArtistInspector.pprint_setters`](https://github.com/matplotlib/matplotlib/blob/43929f31ced180b9ea6c89d25f7ade70d4749b9b/lib/matplotlib/artist.py#L1598)

So in short, not trivial, but probably possible... would need careful consideration of how these pieces interact though (and would include changing some "public" (though potentially should be private) APIs to do it). 

See also #22699 which provides a mechanism for potentially adding/modifying the info that gets put into the table. (It is currently marked as "proof of concept", and has not been merged, though has had some positive response)

File: lib/matplotlib/text.py (hunk 1 above)
Comment (jsalsman) [reply lib/matplotlib/text.py:1005]: But if you don't call it, it is ``left``. How to convey that in the docs?

**Comment by jsalsman** [comment]
> This is a little complicated because these _do_ have a default value in `__init__`, but not in their `set_<attr>` method... but the way the docstring is constructed takes the descriptions from the setters.

@ksunden @StefRe as an interim measure, how would you feel about giving the setters the defaults of those parameters?

File: lib/matplotlib/text.py (hunk 1 above)
Comment (StefRe) [reply lib/matplotlib/text.py]: The default values are (implicitly) documented in the [`__init__`](https://matplotlib.org/stable/api/text_api.html#module-matplotlib.text) method (class signature).

Further, matplotlib [follows](https://matplotlib.org/stable/devel/documenting_mpl.html#writing-docstrings) [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#parameters) conventions:

> When a parameter can only assume one of a fixed set of values, those values can be listed in braces, **with the default appearing first**:
> ```
>order : {'C', 'F', 'A'}
>    Description of `order`.
>```

So the explicit description of the default value is not required in this case, we'd just need to move the `baseline` entry for `verticalalignment` to the first position.

**Comment by StefRe** [comment]
> how would you feel about giving the setter methods the same defaults as those parameters?

I think making an API change just for documentation purposes is not the way to go (also, a setter without an argument isn't a setter anymore, I'd expect such a method be called `reset...`).

File: lib/matplotlib/text.py (hunk 1 above)
Comment (jsalsman) [reply lib/matplotlib/text.py]: > move the `baseline` entry for `verticalalignment` to the first position.

Done.

**Comment by jsalsman** [comment]
> ... probably possible... would need careful consideration of how these pieces interact though (and would include changing some "public" (though potentially should be private) APIs to do it).

@ksunden as @StefRe objected to that, I reverted back to the way he likes it. While setting out along your advice, I came across the mystery as to why the docstring at https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/text.py#L125 doesn't appear anywhere in https://matplotlib.org/devdocs/api/_as_gen/matplotlib.pyplot.text.html -- so that led me to discover the top of that doc page comes from https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/axes/_axes.py#L633 so I'm just going to clarify the alignment parameters there.

**Comment by jsalsman** [comment]
Review please?

**Comment by oscargus** [comment]
Thanks @jsalsman and congratulations on your first merged PR! Hope to see you around!
